src/eval/cache/mod.rs

```diff
@@ -0,0 +1,56 @@
+use super::{Closure, Environment, IdentKind};
```

Comment:
Could deserve a bit of top-level documentation to describe what is the evaluation cache.